cmd/query/integration/token_propagation_test.go

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2019 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// +build token_propagation
+
+package integration
+
+import (
+	"net/http"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const bearerToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJhZG1pbiIsIm5hbWUiOiJKb2huIERvZSIsImlhdCI"
+const bearerHeader = "Bearer " + bearerToken
+
+
+
+func MockingESEToken(t *testing.T, stop *sync.WaitGroup)  {
+	http.HandleFunc("/", func (w http.ResponseWriter, r *http.Request) {
+		authValue := r.Header.Get("Authorization")
+		shouldReceive := "Bearer " + bearerToken
+		assert.Equal(t, shouldReceive, authValue)
+		stop.Done()
+	})
+	http.ListenAndServe(":9200", nil)
+}
+
+func Test_bearTokenPropagationHandler(t *testing.T)  {
+	testCases := []struct {
+		name        string
+		headerValue string
+		headerName  string
+	}{
+		{ name:"Bearer token", headerName:"Authorization", headerValue: bearerHeader},
+		{ name:"Raw Bearer token", headerName:"Authorization", headerValue: bearerToken},
+		{ name:"X-Forwarded-Access-Token", headerName:"X-Forwarded-Access-Token", headerValue: bearerHeader},
+	}
+	stop := &sync.WaitGroup{}
+
+	go MockingESEToken(t, stop)
```

Comment:
Does this need to be a goroutine?

Reply:
Yes, I need to have this running in background mocking the ES, if I don't do this in a goroutine the function will run indefinitely listen on 9200 port, and the next section of the test won't be executed.